### PR TITLE
LA-363 Rename workgroup

### PR DESCRIPTION
**Pull request by dab246**
demo

### CODE REVIEW

**Review by imGok — APPROVED** [review]

File: lib/presentation/widget/shared_space/shared_space_viewmodel.dart [hunk 1]
```diff
@@ -390,6 +392,33 @@ class SharedSpaceViewModel extends BaseViewModel {
...
+      .cancelText(AppLocalizations.of(context).cancel)
+      .onConfirmAction(AppLocalizations.of(context).rename,
+        (value) => store.dispatch(_renameWorkGroupAction(context, value, sharedSpace)))
+      .setErrorString(
+        (value) => getErrorString(context, value))
+      .setTextController(TextEditingController.fromValue(
+      TextEditingValue(
+        text: sharedSpace.name,
+        selection: TextSelection(baseOffset: 0, extentOffset: sharedSpace.name.length)),
+    ))
```
Comment (hoangdat) [inline lib/presentation/widget/shared_space/shared_space_viewmodel.dart:410]: it is messy too understand

File: lib/presentation/widget/shared_space/shared_space_viewmodel.dart [hunk 2]
```diff
@@ -390,6 +392,33 @@ class SharedSpaceViewModel extends BaseViewModel {
...
+        .show(context);
+  }
+
+  OnlineThunkAction _renameWorkGroupAction(BuildContext context, String newName, SharedSpaceNodeNested sharedSpaceNodeNested) {
+    return OnlineThunkAction((Store<AppState> store) async {
+      await _renameWorkGroupInteractor
+        .execute(sharedSpaceNodeNested.sharedSpaceId, RenameWorkGroupRequest(newName, sharedSpaceNodeNested.versioningParameters))
```
Comment (hoangdat) [inline lib/presentation/widget/shared_space/shared_space_viewmodel.dart:417]: if it is too long, please separate each param in each line

Comment (hoangdat) [reply lib/presentation/widget/shared_space/shared_space_viewmodel.dart:417]:
```dart
  .execute(
    sharedSpaceNodeNested.sharedSpaceId, 
    RenameWorkGroupRequest(newName, sharedSpaceNodeNested.versioningParameters))
```

File: lib/presentation/widget/shared_space/shared_space_viewmodel.dart [hunk 3]
```diff
@@ -390,6 +392,33 @@ class SharedSpaceViewModel extends BaseViewModel {
...
+      .onConfirmAction(AppLocalizations.of(context).rename,
+        (value) => store.dispatch(_renameWorkGroupAction(context, value, sharedSpace)))
+      .setErrorString(
+        (value) => getErrorString(context, value))
+      .setTextController(TextEditingController.fromValue(
+      TextEditingValue(
+        text: sharedSpace.name,
+        selection: TextSelection(baseOffset: 0, extentOffset: sharedSpace.name.length)),
+    ))
+        .show(context);
```
Comment (hoangdat) [inline lib/presentation/widget/shared_space/shared_space_viewmodel.dart:411]: no need to handle input error, duplicate name, ...?

File: lib/presentation/widget/shared_space/shared_space_widget.dart [hunk 4]
```diff
@@ -373,6 +374,7 @@ class _SharedSpaceWidgetState extends State<SharedSpaceWidget> {
 
   List<Widget> _contextMenuActionTiles(BuildContext context, SelectableElement<SharedSpaceNodeNested> sharedSpace) {
     return [
+      _renameWorkgroupAction(context, sharedSpace.element),
```
Comment (hoangdat) [inline lib/presentation/widget/shared_space/shared_space_widget.dart:377]: no need to check the role to execute this action or not?

File: lib/presentation/widget/shared_space/shared_space_viewmodel.dart (hunk 3 above)
Comment (dab246) [reply lib/presentation/widget/shared_space/shared_space_viewmodel.dart:411]: I think you should reread my code line by line before making such comments. I have already done it.

File: lib/presentation/widget/shared_space/shared_space_viewmodel.dart (hunk 2 above)
Comment (dab246) [reply lib/presentation/widget/shared_space/shared_space_viewmodel.dart:417]: oke

File: lib/presentation/widget/shared_space/shared_space_viewmodel.dart (hunk 1 above)
Comment (dab246) [reply lib/presentation/widget/shared_space/shared_space_viewmodel.dart:410]: Why? I simply add a TextControlller object to listen for changes in the text in the TextEditer and selectAll text it when it starts to display.

File: lib/presentation/widget/shared_space/shared_space_widget.dart (hunk 4 above)
Comment (dab246) [reply lib/presentation/widget/shared_space/shared_space_widget.dart:377]: oke i fixed it

**Review by hoangdat — APPROVED** [review]

**Pull request merged**